ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java

```diff
@@ -207,10 +213,17 @@ public Optional<TransactionSimulatorResult> processWithWorldUpdater(
     final Account sender = updater.get(senderAddress);
     final long nonce = sender != null ? sender.getNonce() : 0L;
 
-    final long gasLimit =
+    long gasLimit =
         callParams.getGasLimit() >= 0
             ? callParams.getGasLimit()
             : blockHeaderToProcess.getGasLimit();
+    if (rpcGasCap.isPresent()) {
+      final long gasCap = rpcGasCap.get();
+      if (gasCap < gasLimit) {
+        gasLimit = gasCap;
+        LOG.info("Capping gasLimit to " + gasCap);
```

Comment:
What happens when the cap is reached?

Reply:
we should return a -32003 INTRINSIC_GAS_EXCEEDS_LIMIT